test/runtests.jl

```diff
@@ -1,125 +1,311 @@
+
+            result = vcat(f1.(gdf1)...)
+            @test df != result
+            @test size(result) == (24 * 12 * 10, 3)
+            @test !any(ismissing, Tables.matrix(result))
+
+            # Test that we can also mutate the dataframe directly
+            f2.(gdf2)
+            @test result == sort(df, (:hod, :obj))
+        end
     end
 
     @testset "Matrix" begin
+        ctx = Context(; limit=1.0)
+        expected = Matrix(Impute.dropobs(dataset("boot", "neuro"); context=ctx))
         data = Matrix(dataset("boot", "neuro"))
 
         @testset "Drop" begin
-            result = Iterators.drop(data)
+            result = impute(DropObs(; context=ctx), data)
             @test size(result, 1) == 4
+            @test result == Impute.dropobs(data; context=ctx)
+
+            @test result == expected
+            @test Impute.dropobs(data'; vardim=1, context=ctx) == expected'
         end
 
         @testset "Fill" begin
-            result = impute(data, :fill, 0.0; limit=1.0)
+            result = impute(Fill(; value=0.0, context=ctx), data)
             @test size(result) == size(data)
+            @test result == Impute.fill(data; value=0.0, context=ctx)
+
+            data2 = copy(data)
+            Impute.fill!(data2; value=0.0, context=ctx)
+            @test data2 == result
         end
     end
 
     @testset "Not enough data" begin
-        @test_throws ImputeError impute(a, :drop)
+        ctx = Context(; limit=0.1)
+        @test_throws ImputeError impute(DropObs(; context=ctx), a)
+        @test_throws ImputeError Impute.dropobs(a; context=ctx)
     end
 
     @testset "Chain" begin
-        data = Matrix(dataset("boot", "neuro"))
-        result = chain(
-            data,
-            Impute.Interpolate(),
-            Impute.LOCF(),
-            Impute.NOCB();
-            limit=1.0
-        )
-
-        @test size(result) == size(data)
-        # Confirm that we don't have any more missing values
-        @test !any(ismissing, result)
+        orig = dataset("boot", "neuro")
+        ctx = Context(; limit=1.0)
+
+        @testset "DataFrame" begin
+            result = Impute.interp(orig; context=ctx) |> Impute.locf!() |> Impute.nocb!()
+
+            @test size(result) == size(orig)
+            # Confirm that we don't have any more missing values
+            @test !any(ismissing, Matrix(result))
+
+
+            # We can also use the Chain type with explicit Imputor types
+            result2 = impute(
+                Impute.Chain(
+                    Impute.Interpolate(; context=ctx),
+                    Impute.LOCF(),
+                    Impute.NOCB()
+                ),
+                orig,
+            )
+
+            @test result == result2
+        end
+
+        @testset "Column Table" begin
+            result = Tables.columntable(orig) |>
+                Impute.interp!(; context=ctx) |>
+                Impute.locf!() |>
+                Impute.nocb!() |>
+                Tables.matrix
```

Comment:
Is these the same?
```julia
            impute(
                Impute.Chain(
                    Impute.Interpolate(; context=ctx),
                    Impute.LOCF(),
                    Impute.NOCB(),
                ),
                Tables.columntable(orig),
            )

Reply:
(I find `|>` syntax so difficult 😿)

Reply:
It is, but I'm deprecating that syntax in favour of the more functional interface. I think in this case the piping makes it very clear that we're lazily evaluating a chain of imputation functions on the data. I could see reintroducing the `Chain` type if we had explicit performance improvements in mind, but for now it's just another type that doesn't do much. The same behaviour can be handled with less client and package code like this.